libs/libarchfpga/src/physical_types_util.cpp

```diff
@@ -66,79 +63,96 @@ static int get_logical_block_physical_pin_num_offset(t_physical_tile_type_ptr ph
 
 static int get_pin_logical_num_from_pin_physical_num(t_physical_tile_type_ptr physical_tile, int physical_num);
 
-static int get_class_physical_num_from_class_logical_num(t_physical_tile_type_ptr physical_tile,
-                                                         const t_sub_tile* curr_sub_tile,
-                                                         t_logical_block_type_ptr curr_logical_block,
-                                                         int curr_relative_cap,
-                                                         int logical_class_num);
+static std::vector<int> get_pb_pin_src_pins(t_physical_tile_type_ptr physical_type,
+                                            const t_sub_tile* sub_tile,
+                                            t_logical_block_type_ptr logical_block,
+                                            int relative_cap,
+                                            const t_pb_graph_pin* pin);
 
-static std::vector<const t_pb_graph_pin*> get_pb_graph_node_pins(const t_pb_graph_node* pb_graph_node);
+static std::vector<int> get_pb_pin_sink_pins(t_physical_tile_type_ptr physical_type,
+                                             const t_sub_tile* sub_tile,
+                                             t_logical_block_type_ptr logical_block,
+                                             int relative_cap,
+                                             const t_pb_graph_pin* pin);
 
-static std::vector<int> get_pb_pin_driving_pins(t_physical_tile_type_ptr physical_type,
-                                                const t_sub_tile* sub_tile,
-                                                t_logical_block_type_ptr logical_block,
-                                                int relative_cap,
-                                                const t_pb_graph_pin* pin);
+static const t_pb_graph_pin* get_tile_pin_pb_pin(t_physical_tile_type_ptr physical_type,
+                                                 t_logical_block_type_ptr logical_block,
+                                                 int pin_physical_num);
 
-//static std::vector<const t_pb_graph_node*> get_child_pb_graph_node_mode(const t_pb_graph_node* parent_node, int mode_num);
-
-static std::tuple<int, int, int, int> get_pin_index_for_inst(t_physical_tile_type_ptr type, int pin_index, bool is_flat) {
-    int max_ptc = get_tile_ipin_opin_max_ptc(type, is_flat);
-    VTR_ASSERT(pin_index < max_ptc);
+static std::tuple<int, int, int, int, int> get_pin_index_for_inst(t_physical_tile_type_ptr type, int pin_physical_num, bool is_flat) {
```

Comment:
Again, the ourder of elements of the tuple isn't defined. Better to define a struct and return it (more meaningful)